### PR TITLE
Listing page updates

### LINKED ISSUE #31: Individual Listing Modal
Make map pins and listings table contents clickable that render a modal with details for an individual listing. 

This modal should enable users to purchases the listing and receive the seller's contact information.

### LINKED ISSUE #34: Google Map consolidation bubbles
Group pins into consolidated bubbles when zoomed out. Zooming in separates bubbles into pins.

### PR DESCRIPTION
**Pull request by 08martinm**
Listing page now has clickable modals that enable users to express interest in a listing. This stores information to new database table 'Contracts'.

Ironed out bugs regarding viewing new added listings.

edit (casper):
closes #34 
closes #31

### CODE REVIEW

File: client/src/components/Listings/Listing.jsx [hunk 1]
```diff
@@ -1,29 +1,33 @@
 import React from 'react';
-import { Button, Modal } from 'react-bootstrap';
-import { listingPicStyle } from '../../utils/customStyle.js';
 
-export default class Listing extends React.Component {
+export default class ListingTable extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
```
Comment (cassiebaer) [inline client/src/components/Listings/Listing.jsx]: Is this used anywhere?

File: client/src/components/Listings/Listing.jsx [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
+    }
 
+    return (
+      <div id={this.props.index} className="flex-container listingItem" key={this.props.index} style={myStyle} 
+        onMouseOver={this.props.onListingEnter} onMouseLeave={this.props.onListingLeave} onClick={this.props.onListingClick}>
+        <div className="flex-item" id={this.props.index} >{'Item: ' + this.props.listing.name}</div>
+        <div className="flex-item" id={this.props.index} >{'Description: ' + this.props.listing.description}</div>
+        <div className="flex-item" id={this.props.index} >{'Time: ' + this.props.listing.time}</div>
+        <div className="flex-item" id={this.props.index} >{'Reputation: ' + this.props.listing.reputation}</div>
+        <div className="flex-item" id={this.props.index} >{'Price: ' + this.props.listing.price}</div>
```
Comment (cassiebaer) [inline client/src/components/Listings/Listing.jsx]: id's are supposed to be unique

**Comment by cassiebaer** [comment]
🐼

File: server/db/models/contract.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+const Sequelize = require('sequelize');
+const db = require('../connection.js');
+
+let Contract = db.define('contract', {
```
Comment (cassiebaer) [inline server/db/models/contract.js:4]: Where is contract address?

File: server/db/contractModel.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+var db = require('./connection.js');
+var Sequelize = require('sequelize');
+var bluebird = require('bluebird');
+
```
Comment (cassiebaer) [inline server/db/contractModel.js]: this file also exists in `server/db/models/`

File: client/src/utils/utils.js [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-//   return $.ajax({
-//     method: type,
-//     url: url,
-//     data: data,
-//     contentType: "application/json",
-//     success: returnedData => console.log('Successful', type + ':', returnedData),
-//     dataType: 'json'
-//   });
-// };
+export function ajaxJSON(url, type, data) {
```
Comment (cassiebaer) [inline client/src/utils/utils.js:5]: can we use `fetch` so we can drop jQuery as a dep.?

**Comment by cassiebaer** [comment]
👍

**Comment by donovanCarreno** [comment]
:octocat:

**Pull request merged**